soc/arm/nxp_imx/rt/Kconfig.defconfig.series

```diff
@@ -99,6 +98,23 @@ config PM_MCUX_GPC
 	default y if HAS_MCUX_GPC
 	depends on SOC_SERIES_IMX_RT11XX && PM
 
+if SOC_SERIES_IMX_RT10XX && PM
+
+config PM_MCUX_GPC
+	default y if HAS_MCUX_GPC
+
+config PM_MCUX_DCDC
+	default y if HAS_MCUX_DCDC
+
+config PM_MCUX_PMU
+	default y if HAS_MCUX_PMU
+
+# Don't allow SOC to sleep after tests complete when PM is enabled
+config ZTEST
+	select ZTEST_NO_YEILD
```

Comment:
shouldn't this be:

```
config ZTEST_NO_YIELD
    default y if ZTEST
```

Reply:
I chose select because that symbol needs to be enabled unconditionally for RT series SOCs when building with `CONFIG_PM=y and CONFIG_ZTEST=y` .In my view, the build should fail if `CONFIG_ZTEST=y`, `CONFIG_PM=y` and `CONFIG_ZTEST_NO_YEILD=n`, because otherwise twister would proceed to flashing the build, and then the board will effectively be bricked until whoever is running the test recovers it, and the test run would essentially be invalid.

Reply:
After discussing with @hakehuang offline, he said that he would prefer the `default y if ZTEST` approach. Since I made this change with his CI in mind, I've switched the KConfig definition to use `default y if (ZTEST && PM)`. I've also expanded this definition to all RT series parts, not just RT10xx, since twister has issues flashing code to any RT part in low power mode.